UnleashedRecomp/patches/input_patches.cpp

```diff
@@ -403,16 +409,16 @@ PPC_FUNC(sub_8256C938)
         pWorldMapCursor->m_LeftStickVertical = rPadState.LeftStickVertical;
         pWorldMapCursor->m_LeftStickHorizontal = rPadState.LeftStickHorizontal;
 
-        if (rPadState.IsDown(SWA::eKeyState_DpadUp))
+        if (rPadState.IsDown(SWA::eKeyState_DpadUp) && !Config::DisableDPadAsAnalogInput)
```

Comment:
Wrap all of these if statements with a single statement, rather than checking every time.

```cpp
if (!Config::DisableDPadAsAnalogInput)
{
    if (rPadState.IsDown(SWA::eKeyState_DpadUp))
        pWorldMapCursor->m_LeftStickVertical = 1.0f;

    if (rPadState.IsDown(SWA::eKeyState_DpadDown))
        pWorldMapCursor->m_LeftStickVertical = -1.0f;

    if (rPadState.IsDown(SWA::eKeyState_DpadLeft))
        pWorldMapCursor->m_LeftStickHorizontal = -1.0f;

    if (rPadState.IsDown(SWA::eKeyState_DpadRight))
        pWorldMapCursor->m_LeftStickHorizontal = 1.0f;
}
```